File: Java/Object Validator/lib/src/main/java/com/quicksilver/objectvalidator/runtime/Blank.java
package com.quicksilver.objectvalidator.runtime;

import java.util.HashSet;

public class Blank extends Condition {
    public Blank(boolean reversed, String fieldExpression) {
        super(reversed, fieldExpression);
    }

    @Override
    protected boolean isFulfilledBy(Object value, String fullFieldExpression, HashSet<String> passedFields, HashSet<String> failedFields) {
        return switch (value) {
            case null -> throw new RuntimeException("Null values are not supported for 'blank'.");
            case String s -> s.isBlank();
            default -> throw new RuntimeException("Unsupported type for 'blank': " + value.getClass());
        };
    }
}

File: C Sharp/Object Validator/Runtime/Blank.cs
namespace Quicksilver.ObjectValidator.Runtime;
class Blank(bool reversed, string? fieldExpression) : Condition(reversed, fieldExpression)
{
    protected override bool IsFulfilledBy(object? value, string? fullFieldExpression, HashSet<string> passedFields, HashSet<string> failedFields)
    {
        return value switch {
            null => throw new Exception("Null values are not supported for 'blank'."),
            string s => string.IsNullOrWhiteSpace(s),
            _ => throw new Exception("Unsupported type for 'blank': " + value.GetType())
        };
    }
}